half-light.js

```diff
@@ -83,8 +95,11 @@
   Element.prototype.attachShadow = function () {
     let r = old.call(this, ...arguments);
     if (arguments[0].mode !== 'open') return r; 
-    openStylableElements.add(this);
-    Promise.resolve().then(() => {
+    requestAnimationFrame(() => {
+      if (isInDarkRoot(r)) {
```

Comment:
This implementation would need changing if you wanted to share from certain elements down to children. But I don't think that's a big issue as half light in general would probably need to change. You'd probably need a Map of stylesheets with some key and then be able to inherit the entries based on if the key is equal to or below the nearest dark root or not.

Reply:
Yeah, it's independent too, right?  You could have an open root where you wanted to provide styles for your children too... But you'd probably need to figure out an order or a layer for that to go into appropriately too, right? Like, it is more specific than stuff in the head of the doc.  Anyway, that definitely adds a lot of new challenges, and I guess really it only works for stuff you put in your own shadow root - but if you're orchestrating that closely you can probably sort of solve this already... So maybe just fixing the fact that open roots inside of closed ones are getting styled and probably shouldn't be is a bigger deal for more people?

Reply:
Yeah absolutely in agreement there.